src/validation.c

```diff
@@ -195,6 +195,9 @@ validate_netdef_grammar(NetplanNetDefinition* nd, yaml_node_t* node, GError** er
             goto netdef_grammar_error;
     }
 
+    if (nd->ip6_addr_gen_mode > NETPLAN_ADDRGEN_EUI64 && nd->ip6_addr_gen_token)
+        return yaml_error(node, error, "%s: ipv6-address-generation and ipv6-address-token-id are mutually exclusive", nd->id);
+
```

Comment:
I'm uncomfortable with this being an inequality comparison. If other values are added to NetplanAddrGenMode in the future, what guarantees these checks are correct?

Reply:
Good point! I switched to a `!= NETPLAN_ADDRGEN_DEFAULT` comparison, which makes it more explicit to fail if `ipv6-address-generation` is set in any way.